src/main.rs

```diff
@@ -343,6 +343,30 @@ async fn main() -> anyhow::Result<()> {
         .map(|labelpair| parse_label(labelpair))
         .collect::<anyhow::Result<HashMap<String, String>, anyhow::Error>>()
         .context("failed to parse labels")?;
+    let mut labels = HashMap::from([
```

Comment:
So when you declare this variable, you are actually overriding the `labels` value declared on 339. The other thing is that these are the default labels so they should be the ones you start from and then `extend` them with the other user provided data. Generally the order of precedence in most projects is:

1. Command line flags
2. Env vars
3. Defaults

So what you'll probably want to do is start with:

```rust
let mut labels = HashMap::from([
    ("hostcore.arch".into(), ARCH.into()),
    ("hostcore.os".into(), OS.into()),
    ("hostcore.osfamily".into(), FAMILY.into()),
]);
```

Then `.extend` it with the stuff from 339-345